storage/feed.go

```diff
@@ -255,42 +255,99 @@ func (s *Storage) fetchFeedsWithCounters(query string, args ...interface{}) (mod
 	return feeds, nil
 }
 
+const feedByIDQuery = `
+SELECT
+	f.id,
+	f.feed_url,
+	f.site_url,
+	f.title,
+	f.etag_header,
+	f.last_modified_header,
+	f.user_id, f.checked_at at time zone u.timezone,
+	f.parsing_error_count,
+	f.parsing_error_msg,
+	f.scraper_rules,
+	f.rewrite_rules,
+	f.crawler,
+	f.user_agent,
+	f.username,
+	f.password,
+	f.disabled,
+	f.category_id,
+	c.title as category_title,
+	fi.icon_id,
+	u.timezone
+	%s
+FROM feeds f
+LEFT JOIN categories c ON c.id=f.category_id
+LEFT JOIN feed_icons fi ON fi.feed_id=f.id
+LEFT JOIN users u ON u.id=f.user_id
+WHERE
+	f.user_id=$1 AND f.id=$2
+`
+
+// FeedByIDWithWeeklyCount is simialr to FeedByID, but with an additional field of weekly entry count of the feed.
+func (s *Storage) FeedByIDWithWeeklyCount(userID, feedID int64) (*model.Feed, int, error) {
+	var feed model.Feed
+	var iconID interface{}
+	var tz string
+	var weeklyCount int
+	feed.Category = &model.Category{UserID: userID}
+
+	query := fmt.Sprintf(feedByIDQuery,
+		/*addtional_queries=*/ `, (SELECT count(*) FROM entries 
```

Comment:
Why do you still use the "feedByIDQuery"? What I meant in my previous comment is to have a function just to fetch the weekly_count.

Reply:
To reuse codes.
The queries in feedByID and FeedByIDWithWeeklyCount have only one line
difference.
feedByIDQuery is the common part. It does not include the weekly count.
Then we customized it in each function.

Reply:
My point is you can get the weekly count in a separate function, for example `FeedWeeklyCount()`.

Reply:
Then it needs to send another request to the database, right? I want to minimize the traffic between miniflux and the database. Maybe also result in less workload on the database (need benchmark though).

Reply:
Yes, the will create another query to the DB. But I prefer to keep the code simple and stupid for now.

Reply:
Done